pilot/pkg/serviceregistry/serviceentry/conversion.go

```diff
@@ -195,13 +195,24 @@ func convertServices(cfg config.Config) []*model.Service {
 		}
 	}
 
+	allocateAddress := true
+	// donot allocate address for DNS type serviceentry without endpoints specified.
+	if resolution == model.DNSLB && len(serviceEntry.Endpoints) == 0 && serviceEntry.WorkloadSelector == nil {
```

Comment:
Why? I do not understand. `address` and `endpoints` are completely orthogonal. `address` is how we match incoming requests. `endpoints` is where we send outgoing requests

Reply:
The intention here is not to allocate address for ServiceEntry without endpoints, it has nothing to do with address. 

Whether it has address set or not, donot allocate it for the below SE.

```
apiVersion: networking.istio.io/v1alpha3
kind: ServiceEntry
metadata:
  name: jsontime-entry
spec:
  hosts:
  - time.jsontest.com
  ports:
  - number: 80
    name: http-port
    protocol: HTTP
  resolution: DNS
```